### DISCUSSION TITLE
Using .select() to listen to a Map change of a provider, something like StateNotifier<Map<String, List<Movie>>>

### DISCUSSION THREAD

**Discussion by ad-on-is** [rrousselGit/riverpod#2643] [Q&A]
I have a column with a dynamic number of lists.

Popular Movies
[][][][][][][][][]
..................................................
Trending Movies
[][][][][][][][][]
..................................................

etc...

Now I want to add a pagination to each of the lists. And I could not find a way to do that, especially since every list uses the same provider, but calling a different endpoint, like /movies/trending .. /movies/popular, etc...

I know this is wrong, but my provider looks something like this.

```
class ItemPagination {
  int page = 1;
  List<Trakt> items = [];

  ItemPagination({this.page = 1, this.items = const []});
}

class ItemPaginationProvider extends StateNotifier<Map<String, ItemPagination>> {
  final Ref ref;
  final TraktService traktService;

  ItemPaginationProvider(this.ref, this.traktService) : super({});

  void init(String url) async {
    if (state[url] == null) {
      state[url] = ItemPagination();
    }
    state[url]!.items = await traktService.getItems(url);

  }

  void next(String url) async {
    state[url]!.page++;
    state[url]!.items.addAll(await traktService.getItems(url));
  }
}

final paginationProvider =
    StateNotifierProvider((ref) => ItemPaginationProvider(
          ref,
          ref.watch(traktProvider),
        ));

```

Now, each of my widgets would need something along the lines
`ref.watch(paginationProvider.select((value)=>value[url] == widget.url))`   or something like that.

The closest thing I could find was https://github.com/rrousselGit/riverpod/pull/1532 but this does not deal with dynamic maps.

Do I have to create a dedicated provider for each of the lists, or is there a simple more dynamic way?

**Comment by rrousselGit** [comment]
> `ref.watch(paginationProvider.select((value)=>value[url] == widget.url))`

That line doesn't really make sense. You'd get a `bool` back, which is unlikely to be what you want.

Instead you probably want to do:

```dart
ref.watch(paginationProvider.select((value)=>value[widget.url]));
```

**Reply by ad-on-is** [reply]
Thanks, but I solved it with .family() and List<Movie> instead of Map<String, <List<Movie>>> which also does exactly what I want, and might even be the more proper way of doing so with AutoDisposeStateNotifierProviderFamily, right?

**Reply by rrousselGit** [reply]
Sure, that's a good solution too
